pkg/virt-controller/watch/pool.go

```diff
@@ -605,30 +621,102 @@ func injectHashIntoVM(vm *virtv1.VirtualMachine, vmHash string, vmiHash string)
 		vm.Spec.Template.ObjectMeta.Labels = map[string]string{}
 	}
 
-	vm.Labels[virtv1.VirtualMachineTemplateHash] = vmHash
-	vm.Spec.Template.ObjectMeta.Labels[virtv1.VirtualMachineInstanceTemplateHash] = vmiHash
+	vm.Labels[virtv1.VirtualMachinePoolRevisionName] = revisionName
+	vm.Spec.Template.ObjectMeta.Labels[virtv1.VirtualMachinePoolRevisionName] = revisionName
```

Comment:
Should this be an annotation on the VMI? No concerns on the VM. I just wonder about `expose` commands which would automatically consider all labels (we may filter out kubevirt.io prefixed labels though, let's double check).

Reply:
fixed